models/nc-news4.model.js

```diff
@@ -1,13 +1,24 @@
 const db = require("../db/connection.js");
 
 const getArticlesById = (article_id) => {
+  if (typeof article_id === "undefined" || !article_id.match(/^\d+$/)) {
+    return Promise.reject({
+      status: 400,
+      message: "Bad request",
+    });
+  }
+
```

Comment:
The above works, but both of these conditions can be caught by PSQL itself. If the article id isn't a number and the query below was attempted, PSQL would error out and wouldn't be able to fulfil the query. In this case, we can catch the error back in our controller and then pass it to error handlers using next. Then I would check for a code under err.code and use the code to send a response to the client :) 

There's nothing _wrong_ with how you've implemented this, but we can let PSQL handle it for us. We just need to catch the error and do something with it!